### ISSUE TITLE
The problem of segmenting trace

### ISSUE THREAD

**Issue by Gitxg** [google/perfetto#656]
After using the split tool to split the trace, the segmented file cannot be opened from the second one, resulting in an error message.

The segmentation command I am using is
split -b 50M trace.perfetto-trace

![1](https://github.com/google/perfetto/assets/50760172/d120d21d-336e-435c-8885-e0b93810dc40)

May I ask how to divide a large trace file into several small files? I think such a tool is very necessary.
Thanks, Best wishes.

**Comment by primiano** [comment]
You can't just randomly split a file and expect it to get opened.

There are two things here:

1) If you really want to split it, you have to do respecting the protobuf message boundaries. You can use the `synchronization_marker`, looking for the 128-bit UID `{0x82, 0x47, 0x7a, 0x76, 0xb2, 0x8d, 0x42, 0xba, 0x81, 0xdc, 0x33,0x32,0x6d, 0x57, 0xa0, 0x79}` and split on that boundary.

2) Even if you manage to successfully split a trace, you'll find that many threads/processes don't have names, because some data gets emitted only once in the context of a trace. If you split it away, important information to decode the rest of the trace will be missing.

**Comment by Gitxg** [comment]
The protobuf message boundaries and other information are too professional. For performance problem analysts, it may be more necessary to have an official cutting tool, even if some of the references cannot be obtained after cutting, and there are problems with the process you described.
Can it be developed as a new requirement to create a segmentation tool.
Thank you, best wishes

**Comment by LalitMaganti** [comment]
Instead of collecting a single large trace, have you considered collecting many smaller traces instead? This is what we do in Google and it works pretty well.

In general, building a tool which can split traces is not a priority for us. Patches for doing this are welcome but will not be something we get to ourselves. Alternatively, it might be worth doing this in some independent project.

**Comment by primiano** [comment]
> The protobuf message boundaries and other information are too professional. For performance problem analysts, it may be more necessary to have an official cutting tool, even if some of the references cannot be obtained after cutting, and there are problems with the process you described. Can it be developed as a new requirement to create a segmentation tool. Thank you, best wishes

The what I'd suggest is:

- Load the whole big trace in trace processor (Will take a while), maybe via the python API
- Export segments of data from the tables you need and pass them to your analysts as csv files or panda dataframes
